### PR TITLE
chore: vicky custom metrics endpoint + port

### PR DESCRIPTION
**Pull request by brodeyn**

### CODE REVIEW

File: modules/gradient-processing/input.tf [hunk 1]
```diff
@@ -357,6 +357,26 @@ variable "victoria_metrics_vmcluster_service_endpoint" {
   default = "http://vmselect-gradient-processing-victoria-metrics:8481/select/0/prometheus"
 }
 
+variable "victoria_metrics_vmsingle_service_metrics_adapter_endpoint" {
```
Comment (bbatha) [inline modules/gradient-processing/input.tf:360]: Could you get rid of `gradient_metrics_conn_str` by appending these strings together in a local?

Comment (brodeyn) [reply modules/gradient-processing/input.tf:360]: I really don't want to be doing slice and joins in terraform tbh... I'd have to slice the path off and append the port and then re-append the path.

Comment (brodeyn) [reply modules/gradient-processing/input.tf:360]: these variables live next to each other so they should be easy to update together

Comment (bbatha) [reply modules/gradient-processing/input.tf:360]: Do not do that we have far too many variables in this terraform many of them duplicate values in different formats. Its less code, then even just adding these new vars, to just do the string concatenation in terraform anyway:

```
variable "victoria_metrics_host" {
    type = string
    default = "vmsingle-gradient-processing-victoria-metrics"
}

variable "victoria_metrics_port" {
    type = number
    default = 8429
}

variable "victoria_metrics_path" {
   type = string
   default = "/prometheus"
}

locals {
     gradient_metrics_conn_str = "http://${var.victoria_metrics_host}:${var.victoria_metrics_port}/${var.victoria_metrics_path}"
     metrics_url_no_port = "http://${var.victoria_metrics_host}/${var.victoria_metrics_path}"
}
```

**Review by bbatha — APPROVED** [review]

**Pull request merged**